chunky/src/java/se/llbit/chunky/renderer/scene/SampleBuffer.java

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.scene;
+
+import se.llbit.math.ColorUtil;
+
+public class SampleBuffer {
+  public final int width;   // = rowSize/3, used for alpha
+  public final int rowSize; // = width * 3, used for samples
+
+  public final int height;   // = rowCount
+  public final int rowCount; // = height
+
+  protected final double[][] samples;
+  // protected int[][] spp;     // for non-uniform sample distribution
```

Comment:
As mentioned above, until we have that, maybe store the spp here, i.e. `protected int spp` to simplify the `addSample` method.

Reply:
Next commit is partial implementation of this, though `int Scene#spp` still exists and is used in most locations.